tests/data_context/test_data_context_config_errors.py

```diff
@@ -9,22 +13,34 @@
 BASE_DIR = "fixtures"
 
 
-def test_DataContext_raises_error_on_config_not_found():
-    local_dir = file_relative_path(__file__, os.path.join(BASE_DIR, ""))
+@pytest.fixture
+def build_local_dir(tmp_path):
+    def _build_local_dir(fixture_subdir: str | None = None) -> pathlib.Path:
+        local_dir = tmp_path / "root"
+        if fixture_subdir:
+            fixture_path = file_relative_path(
+                __file__, os.path.join(BASE_DIR, fixture_subdir)
+            )
+            shutil.copytree(fixture_path, local_dir)
+        return local_dir
+
+    return _build_local_dir
+
+
+def test_DataContext_raises_error_on_config_not_found(build_local_dir):
+    local_dir = build_local_dir()
     with pytest.raises(gx_exceptions.ConfigNotFoundError):
         get_context(context_root_dir=local_dir)
 
 
-def test_DataContext_raises_error_on_unparsable_yaml_file():
-    local_dir = file_relative_path(__file__, os.path.join(BASE_DIR, "bad_yml"))
+def test_DataContext_raises_error_on_unparsable_yaml_file(build_local_dir):
```

Comment:
Need to make sure we aren't writing to disk and are always using `tmp_path` (a function scoped `pathlib.Path` obj)